x/evm/keeper/statedb.go

```diff
@@ -159,7 +164,22 @@ func (k *Keeper) SetState(ctx sdk.Context, addr common.Address, key common.Hash,
 	)
 }
 
+// SetCodeHash sets the code hash for the given contract address and deletes an existing
+// entry if the passed code hash is empty.
+func (k *Keeper) SetCodeHash(ctx sdk.Context, addr common.Address, codeHash common.Hash) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixCodeHash)
+	key := addr.Bytes()
+	hashBytes := codeHash.Bytes()
+	if types.BytesAreEmptyCodeHash(hashBytes) {
+		store.Delete(key)
+	} else {
+		store.Set(key, hashBytes)
+	}
+}
+
 // SetCode set contract code, delete if code is empty.
+//
+// TODO: do we still need this separation of the codehash and the code itself if both is stored in the same keeper?
```

Comment:
same question as posed above 👆

Reply:
This comes from geth, so I think we have to keep it

Reply:
I believe also that makes sense since in this way you can just retrieve the hash without reading the possibly big code of a contract.